File: take-home-java-billing/src/test/java/com/ppro/interview/billing/HelloWorldControllerTest.java
package com.ppro.interview.billing;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
import org.springframework.test.web.servlet.MockMvc;

import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;

@WebMvcTest(HelloWorldController.class)
class HelloWorldControllerTest {

    @Autowired
    private MockMvc mockMvc;

    @Test
    void sayHiToKitten() throws Exception {
        mockMvc.perform(get("/v1/hello/kitten"))
                .andExpect(status().isOk())
                .andExpect(content().json("{ \"message\": \"hello\", \"recipient\": \"kitten\" }", true));
    }

    @Test
    void returnsNotFoundIfNoNameProvided() throws Exception {
        mockMvc.perform(get("/v1/hello/")).andExpect(status().isNotFound());
    }
}

File: take-home-kotlin-billing/src/test/kotlin/com/ppro/interview/billing/HelloWorldControllerTest.kt
package com.ppro.interview.billing

import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest
import org.springframework.test.web.servlet.MockMvc
import org.springframework.test.web.servlet.get
import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status

@WebMvcTest(HelloWorldController::class)
internal class HelloWorldControllerTest {

    @Autowired
    private lateinit var mockMvc: MockMvc

    @Test
    fun sayHiToKitten() {
        mockMvc.get("/v1/hello/kitten").andExpect {
            status().isOk
            content { json("""{ "message": "hello", "recipient": "kitten" }""", strict = true) }
        }
    }

    @Test
    fun returnsNotFoundIfNoNameProvided() {
        mockMvc.get("/v1/hello/").andExpect { status().isNotFound }
    }
}